src/dbnode/storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -1031,7 +1104,13 @@ func (s *fileSystemSource) bootstrapFromIndexPersistedBlocks(
 		// as we've already passed the ranges fulfilled to the block that
 		// we place in the IndexResuts with the call to Add(...).
 		res.result.index.Add(indexBlockByVolumeType, nil)
```

Comment:
F/u on not adding non default vol type index segments to results if we are planning on deleting them.

Reply:
After sync, since this is a corner case... the cleanup logic is simpler to do when we actually persist an index block to disk (albeit higher overhead due to opening and closing index segments).